torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -9725,6 +9725,47 @@ def forward(self, inp):
                 ddp._check_reducer_finalized()
                 ddp(input)
 
+        @skip_if_lt_x_gpu(2)
+        @skip_but_pass_in_sandcastle_if(
+            BACKEND not in DistTestCases.backend_feature["ddp"],
+            f"The {BACKEND} backend does not support DistributedDataParallel",
+        )
+        def test_nccl_init_abort(self):
+            """
+            Tests that we can abort a NCCL communicator during initialization and
+            recover appropriately.
+            """
+            # Reinitialize global process group with TORCH_NCCL_USE_COMM_NONBLOCKING=1
+            os.environ["TORCH_NCCL_USE_COMM_NONBLOCKING"] = "1"
```

Comment:
Would we want to add a mechanism to clear this setting? Right now it would be a sticky change---unfortunately I think it would require adding another mechanism to override the environment variable.

Reply:
Actually this code is being run in the child process of the unit test and hence should not affect any other tests. Basically the distributed tests run each test method by spawning N children and running the test methods in those N children.

Once the test is done the children are terminated and a new set of processes are started for the subsequent tests.